### ISSUE TITLE
NewImageFromImage causes panic: buffered: the command queue is not available yet at Pixels

### ISSUE THREAD

**Issue by kyeett** [hajimehoshi/ebiten#1295]
short error:

```
panic: buffered: the command queue is not available yet at Pixels
```

```go
package main

import (
	"fmt"
	"image"
	"log"

	"github.com/hajimehoshi/ebiten"
	"github.com/hajimehoshi/ebiten/ebitenutil"
	"github.com/peterhellberg/gfx"
)

type game struct{}

func init() {
	img := gfx.MustOpenImage("noise.png")
	baseImage, err := ebiten.NewImageFromImage(img, ebiten.FilterDefault)
	if err != nil {
		log.Fatal()
	}

	thingV1 := baseImage.SubImage(image.Rect(0, 0, 8, 8).Add(image.Pt(8*1, 8*0))).(*ebiten.Image)
	thingImage, _ := ebiten.NewImageFromImage(thingV1, ebiten.FilterDefault)
	fmt.Println(thingImage)
}

func (g game) Update(_ *ebiten.Image) error {
	return nil
}

func (g game) Draw(screen *ebiten.Image) {
	ebitenutil.DebugPrint(screen, "template")
}

const (
	screenWidth  = 800
	screenHeight = 600
)

func (g game) Layout(w, h int) (int, int) {
	return screenWidth, screenHeight
}

func main() {
	g := &game{}
	if err := ebiten.RunGame(g); err != nil {
		log.Fatal(err)
	}
}

```

Long error
```
panic: buffered: the command queue is not available yet at Pixels

goroutine 1 [running, locked to thread]:
github.com/hajimehoshi/ebiten/internal/buffered.checkDelayedCommandsFlushed(...)
	/Users/test/go/src/github.com/hajimehoshi/ebiten/internal/buffered/command.go:90
github.com/hajimehoshi/ebiten/internal/buffered.(*Image).Pixels(0xc000060480, 0x8, 0x0, 0x1, 0x1, 0x20300000000000, 0x4f3ffff, 0x4f61668, 0x10010004f3ffff, 0x0)
	/Users/test/go/src/github.com/hajimehoshi/ebiten/internal/buffered/image.go:135 +0x4e5
github.com/hajimehoshi/ebiten/internal/mipmap.(*Mipmap).Pixels(...)
	/Users/test/go/src/github.com/hajimehoshi/ebiten/internal/mipmap/mipmap.go:90
github.com/hajimehoshi/ebiten.(*Image).At(0xc0000b4000, 0x8, 0x0, 0xc0000acc00, 0x411f835)
	/Users/test/go/src/github.com/hajimehoshi/ebiten/image.go:722 +0xce
image/draw.drawRGBA(0xc0000b40c0, 0x0, 0x0, 0x8, 0x8, 0x444d420, 0xc0000b4000, 0x8, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/image/draw/draw.go:519 +0x1e8
image/draw.DrawMask(0x444e4a0, 0xc0000b40c0, 0x0, 0x0, 0x8, 0x8, 0x444d420, 0xc0000b4000, 0x8, 0x0, ...)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/image/draw/draw.go:180 +0x982
image/draw.Draw(...)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/image/draw/draw.go:101
github.com/hajimehoshi/ebiten.imageToBytesSlow(0x444d420, 0xc0000b4000, 0x0, 0x10, 0x8)
	/Users/test/go/src/github.com/hajimehoshi/ebiten/imagetobytes.go:88 +0x213
github.com/hajimehoshi/ebiten.imageToBytes(0x444d420, 0xc0000b4000, 0x0, 0xc0000b6030, 0x8)
	/Users/test/go/src/github.com/hajimehoshi/ebiten/imagetobytes.go:74 +0x39f
github.com/hajimehoshi/ebiten.NewImageFromImage(0x444d420, 0xc0000b4000, 0x0, 0x10, 0x8, 0x444d420)
	/Users/test/go/src/github.com/hajimehoshi/ebiten/image.go:850 +0x112
main.init.0()
	/Users/test/go/src/github.com/kyeett/games/examples/troubleshooting/main.go:23 +0xdd
exit status 2
```

**Comment by hajimehoshi** [comment]
Thanks!

Further minimized case:

```
package main

import (
        "github.com/hajimehoshi/ebiten"
)

func init() {
        img, _ := ebiten.NewImage(10, 10, ebiten.FilterDefault)
        ebiten.NewImageFromImage(img, ebiten.FilterDefault)
}

func main() {
}
```

This is because NewImageFromImage tries to call `At` of the given image, which is forbidden before the main loop.

**Comment by hajimehoshi** [comment]
Let me mark this as 'won't fix'.

* NewImageFromImage invokes `At` of the given image, but this cannot be called before the main loop especially due to Android.
* Giving an Ebiten image to NewImageFromImage is not so useful. You can copy Ebiten's image by ReplacePixels or DrawImage if you want.

**Comment by muchitto** [comment]
Just curious, why is this forbidden (I just ran into the same issue with the same setup)? I wanted to do some pixel level manipulation of the image after loading it, but before I draw it to the screen. :)

I'm loading an image with `NewImageFromFile` function in an `init` function before I call `RunGame`, but I cannot call `At` of that loaded image.

**Comment by hajimehoshi** [comment]
Good question. This is for portability. On some environments like Android, OpenGL cannot be initialized before the main loop. (Ebiten's Android application works as a shared library, and when `init` function is called, there is no guarantee that OpenGL functions are available)

To solve this issue, I think I'd need to implement a software rendering instead.

BTW, instead of `NewImageFromImage`, you can copy the `ebiten.Image` by `NewImage` and `DrawImage`.

**Comment by muchitto** [comment]
Ok, so just checking, there isn't a way to check if the OpenGL functions are available and then run a `Init` function once (where I do all of my image manipulation before using them), before starting to run all the `Update` and `Draw` functions?

**Comment by hajimehoshi** [comment]
No, there is no way to detect whether OpenGL is available.

**Comment by muchitto** [comment]
Thanks for the help. Btw, this library is a joy to use, thanks for making it!
